### PR TITLE
Krithiga/integration microbenchmark

### PR DESCRIPTION
**Pull request by krithi96**
- launch scripts for microbenchmark using gem5art integrated with python script for microbenchmarks.

### CODE REVIEW

File: docs/configs-microbenchmark-test/run_config1.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
```
Comment (powerjg) [inline docs/configs-microbenchmark-test/run_config1.py:1]: Can you give this file a more descriptive name?

File: docs/configs-microbenchmark-test/run_config1.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    bandwidth = '0B/s'
+
+# Add more Memories under test before this
+valid_memories = [InfMemory, SingleCycleMemory, SlowMemory]
+valid_memories = {cls.__name__[:-6]:cls for cls in valid_memories}
+parser = argparse.ArgumentParser()
+parser.add_argument('cpu', choices = valid_cpus.keys())
+parser.add_argument('memory_model', choices = valid_memories.keys())
+parser.add_argument('binary', type = str, help = "Path to binary to run")
+parser.add_argument('--l1i_size',type = str,  help = "CachesizeofL1i")
```
Comment (powerjg) [inline docs/configs-microbenchmark-test/run_config1.py:193]: Do we need these optional arguments? If not, let's remove them.

File: docs/configs-microbenchmark-test/run_config1.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+parser.add_argument('--l1d_size',type = str,  help = "CachesizeofL1d")
+parser.add_argument('--l2_size', type = str,  help = "CachesizeofL2")
+
+args  = parser.parse_args()
+
+    #parser.add_argument('binary_input', type = str, help = "Inputs to the binary")
```
Comment (powerjg) [inline docs/configs-microbenchmark-test/run_config1.py:199]: Let's remove all of the commented out code, if it's not needed.

File: docs/configs-microbenchmark-test/run_config1.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+command_line_args(args.l1i_size,args.l1d_size,args.l2_size)
```
Comment (powerjg) [inline docs/configs-microbenchmark-test/run_config1.py:201]: This is weird... it has a bad code smell (https://en.wikipedia.org/wiki/Code_smell). Let's talk about this in person.

File: docs/configs-microbenchmark-test/run_config1.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+if exit_event.getCause() != 'exiting with last active thread context':
+    print("Benchmark failed with bad exit cause.")
+    print(exit_event.getCause())
+    exit(1)
+if exit_event.getCode() != 0:
+    print("Benchmark failed with bad exit code.")
+    print("Exit code {}".format(exit_event.getCode()))
+    exit(1)
+
+print("{} ms".format(m5.curTick()/1e9))
```
Comment (powerjg) [inline docs/configs-microbenchmark-test/run_config1.py:222]: There should be a function in `m5` which will do this conversion from ticks to ms. Using 1e9 could be wrong if the tick rate changed (this won't happen in practice, but it theoretically possible).

File: docs/launch_microbench_test.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+import os
+import sys
+from uuid import UUID
+
+from gem5art.artifact.artifact import Artifact
+from gem5art.run import gem5Run
+from gem5art import tasks
+
+"""packer = Artifact.registerArtifact(
```
Comment (powerjg) [inline docs/launch_microbench_test.py:13]: Two problems here :)
1) `'''` is not a good way to comment in python.
2) We should remove all commented code, if it's not needed.

File: docs/launch_microbench_test.py [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+                documentation = f"Kernel binary for {version} with simple "
+                                 "config file",
+            )
+    for version in linuxes
+}"""
+
+if __name__ == "__main__":
+    #boot_types = ['init']#, 'systemd']
+    num_cpus = ['1']#, '2', '4', '8']
+    cpu_types = ['kvm']#, 'atomic', 'simple', 'o3']
```
Comment (powerjg) [inline docs/launch_microbench_test.py:104]: This can't be right....

File: docs/launch_microbench_test.py [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+    
+    #For configuration 3:
+    #Cache_type:
+    cache_type = 'L1_cache' #'L2_cahe'
+    #L1Cache_sizes.
+    L1D = ['32kB','128kB','64kB']
+    #L2Cache_sizes.
+    L2C = ['1MB','512kB']
+
+    if config == 'config1':
```
Comment (powerjg) [inline docs/launch_microbench_test.py:137]: We need to come up with a better way to code all of this :).

File: docs/launch_microbenchmark.py [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+import os
+import sys
+from uuid import UUID
+
+from gem5art.artifact.artifact import Artifact
+from gem5art.run import gem5Run
+from gem5art.tasks.tasks import run_gem5_instance
+
+"""packer = Artifact.registerArtifact(
```
Comment (powerjg) [inline docs/launch_microbenchmark.py:13]: Ditto the other file.

**Review by powerjg — COMMENTED** [review]
Thanks, Krithiga! Even though this was merged, I went ahead and did a post commit review. If you could address these things in another PR, that would be great.

**Comment by krithi96** [comment]
Hello jason,

I am really sorry, I didn't mean to merge it. I was poking around in github
and did it by mistake. I only wanted push and do a PR to
launch_microbenchmark.py, I am still getting used to git and this happened
bymistake when I was doing a PR, I will make a PR after cleaning these up,
so sorry again.

Regards
Krithiga

**Comment by powerjg** [comment]
No worries! Post commit review is just as good!

**Pull request merged**